src/LinAlg/hiopVectorPar.cpp

```diff
@@ -124,6 +125,22 @@ void hiopVectorPar::setToConstant(double c)
 {
   for(int i=0; i<n_local_; i++) data_[i]=c;
 }
+
+void hiopVectorPar::set_to_random_constant(double minv, double maxv)
+{
+  std::uniform_real_distribution<double> unif(minv,maxv);
+  std::default_random_engine re;
+  double a_random_double;
+
+#ifdef NDEBUG
+  re.seed(std::chrono::system_clock::now().time_since_epoch().count());
+#endif
+
+  for(index_type i=0; i<n_local_; ++i) {
+    data_[i] = unif(re);
+  }
+}
+
```

Comment:
Any good way to implement this function in hiopVectorRAJAPar?
@cnpetra @ashermancinelli

Reply:
cuRand is shipped in cuda toolkit, but we might have to add some linker flags.

Reply:
[We could do almost exactly what this Thrust example is doing with RAJA](https://docs.nvidia.com/cuda/curand/device-api-overview.html#thrust-and-curand-example).

Reply:
@ashermancinelli I know we have cuRAND and rocRAND, but the problem now is that in this case I cannot see a portable solution to both cuda and hip. It seems to me that we need to use `#ifdef HIOP_USE_CUDA` with `cuRAND`, and `#ifdef HIOP_USE_HIP` with `rocRAND`.

Reply:
I see now. Grepping through raja's docs doesn't find anything for random, so I think ugly macros are the only way to go.

Reply:
not much to add -- it appears we need to rely on roc and cuda api (with `ifdef`s). We will add an abstraction layer for device math functions call later on.